src/kcas.ml

```diff
@@ -153,29 +153,29 @@ let overlap () = failwith "kcas: location overlap" [@@inline never]
 
 type splay = Miss : splay | Hit : 'a loc * 'a state -> splay
 
-let rec splay x = function
+let rec splay ~hit_parent x = function
```

Comment:
The `hit_parent` flag allows `splay` to be used for both plain lookups and updates (insert or replace).  It means that in case the key is not found, the node that would have been the parent is brought to the root.  With a splay tree each access needs to update the tree &mdash; otherwise it is possible to repeat a slow access and the amortized `O(n*log(n))` complexity no longer applies.